### ISSUE TITLE
activation function

### ISSUE THREAD

**Issue by zfhupc** [ymishin-gmu/LAMMPS-USER-PINN#2]
Hello, Professor Y.Mishin and Dr. James Hickman，
May I ask what activation function you are using when building PINNs

**Comment by ymishin-gmu** [comment]
Our training code has a choice 3 activation functions:
case 0: Sigmoid  {1/1+e−x}
case 1: Sigmoid-1/2 { 1/2tanh (x/2) }
case 2: tangent hyperbolic {tanh(x)}
Different potentials use different functions.

**Comment by ymishin-gmu** [comment]
> Hello, Professor Y.Mishin and Dr. James Hickman，
> May I ask what activation function you are using when building PINNs

Our training code has a choice 3 activation functions:
case 0: Sigmoid {1/1+e−x}
case 1: Sigmoid-1/2 { 1/2tanh (x/2) }
case 2: tangent hyperbolic {tanh(x)}
Different potentials use different functions.

**Comment by zfhupc** [comment]
> > Hello, Professor Y.Mishin and Dr. James Hickman，
> > May I ask what activation function you are using when building PINNs
> 
> Our training code has a choice 3 activation functions: case 0: Sigmoid {1/1+e−x} case 1: Sigmoid-1/2 { 1/2tanh (x/2) } case 2: tangent hyperbolic {tanh(x)} Different potentials use different functions.

I am very glad to receive your reply. I tried to reproduce your article about PINN with pytorch. I have two questions that I would like to ask you. The first is how you are dealing with data with different atomic numbers, I cannot train them in one batch. The second is the source code of the formula given in your paper, can you provide it, I want to verify the code I rewrote in matrix form.

**Comment by ymishin-gmu** [comment]
The PINN model has been generalized to multicomponent materials (different atomic numbers) in 
Y. Mishin, “Machine-learning interatomic potentials for materials science”, Acta Materialia 214, 116980 (2020).
[doi.org/10.1016/j.actamat.2021.116980](https://doi.org/10.1016/j.actamat.2021.116980)]. 

We routinely train binary potentials in the batch mode using our C++/MPI code, not pytorch.

Our training code is a home-made creature with many hard-wired knobs and levers and we only use it inside the group.
Other useful papers to look at are:

Yi-Shen Lin, Ganga P. Purja Pun and Yuri Mishin, “Development of a physically-informed neural network interatomic potential for tantalum”, Computational Materials Science 205, 111180 (2022).
G. P. Purja Pun, V. Yamakov, J. Hickman, E. H. Glaessgen, and Y. Mishin, “Development of a general-purpose machine-learning interatomic potential for aluminum by the physically informed neural network method”, Physical Review Materials 4, 113807 (2020).

Good luck!
